esphome/components/debug/debug_zephyr.cpp

```diff
@@ -322,6 +326,21 @@ void DebugComponent::get_device_info_(std::string &device_info) {
 #endif
   }
 #endif
+  auto uicr = [](volatile uint32_t *data, uint8_t size) {
+    std::string res;
+    char buf[sizeof(uint32_t) * 2 + 1];
+    for (size_t i = 0; i < size; i++) {
+      if (i > 0) {
+        res += ' ';
+      }
+      sprintf(buf, "%08X", data[i]);
+      res += buf;
```

Comment:
```suggestion
      res += format_hex_pretty<uint32_t>(data[i], '\0', false);
```

Maybe better.. its a bit safer. All the string concat is going to generate a lot of code but its for debug anyways ...